frontend/dockerfile/dockerfile2llb/convert.go

```diff
@@ -135,26 +135,86 @@ func ListTargets(ctx context.Context, dt []byte) (*targets.List, error) {
 	return l, nil
 }
 
+type namedContextState struct {
+	State llb.State
+	Image *image.Image
+
+	// State used for initializing/re-initializing the state.
+	name         string
+	out          *mutableOutput
+	copt         dockerui.ContextOpt
+	includePaths map[string]struct{}
+}
+
+// newNamedContextState returns an uninitialized namedContextState.
+func newNamedContextState(ctx context.Context, client *dockerui.Client, name string, copt dockerui.ContextOpt) (*namedContextState, error) {
+	out := &mutableOutput{}
+	nc := &namedContextState{
+		State: llb.NewState(out),
+		name:  name,
+		out:   out,
+		copt:  copt,
+	}
+	return nc.reinit(ctx, client)
+}
+
+// reinit will reinitialize this namedContextState. If it is successful, it will return itself.
+// If the new state would be invalid, it does not modify the current context and it returns nil along with
+// an error if one was present.
+func (nc *namedContextState) reinit(ctx context.Context, client *dockerui.Client) (*namedContextState, error) {
+	// Force a copy of the context options.
+	copt := nc.copt
+	if len(nc.includePaths) > 0 {
+		if includePatterns := normalizeContextPaths(nc.includePaths); includePatterns != nil {
+			copt.LocalOpts = append(copt.LocalOpts, llb.FollowPaths(includePatterns))
```

Comment:
`copt.LocalOpts` will get modified on the original here, which I imagine isn't the intent (given the above comment)?